src/ResourceManager/Websites/Commands.Websites/Cmdlets/BackupRestore/RestoreAzureWebAppSnapshot.cs

```diff
@@ -0,0 +1,77 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.WebApps.Utilities;
+using Microsoft.Azure.Management.WebSites.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.BackupRestore
+{
+    /// <summary>
+    ///     Restores an Azure Web App snapshot
+    /// </summary>
+    [Cmdlet(VerbsData.Restore, "AzureRmWebAppSnapshot")]
+    public class RestoreAzureWebAppSnapshot : WebAppOptionalSlotBaseCmdlet
+    {
+        [Parameter(Position = 3, Mandatory = true, HelpMessage = "The timestamp of the snapshot.", ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string SnapshotTime;
```

Comment:
@Nking92 do you think it would be easier for the user to use this parameter if it was a `DateTime` rather than a `string`?

Reply:
Yes, that's a good suggestion. I will convert it to a string afterward.